src/platforms/dotnet/guides/aspnet/performance/included-instrumentation.mdx

```diff
@@ -0,0 +1,142 @@
+            // Start a transaction that encompasses the current request
+            var transaction = Context.StartSentryTransaction();
+
+            // Attach transaction to the request context to finish it when the request ends
+            Context.Items["__SentryTransaction"] = transaction;
+        }
+
+        protected void Application_EndRequest()
+        {
+            // Finish the currently active transaction
+            if (Context.Items.Contains("__SentryTransaction"))
+            {
+                var transaction = Context.Items["__SentryTransaction"] as ISpan;
+
+                var status =
+                    Context.Response.StatusCode < 400 ? SpanStatus.Ok :
+                    Context.Response.StatusCode == 400 ? SpanStatus.InvalidArgument :
+                    Context.Response.StatusCode == 401 ? SpanStatus.Unauthenticated :
+                    Context.Response.StatusCode == 403 ? SpanStatus.PermissionDenied :
+                    Context.Response.StatusCode == 404 ? SpanStatus.NotFound :
+                    Context.Response.StatusCode == 409 ? SpanStatus.AlreadyExists :
+                    Context.Response.StatusCode == 429 ? SpanStatus.ResourceExhausted :
+                    Context.Response.StatusCode == 499 ? SpanStatus.Cancelled :
+                    Context.Response.StatusCode < 500 ? SpanStatus.InvalidArgument :
+                    Context.Response.StatusCode == 500 ? SpanStatus.InternalError :
+                    Context.Response.StatusCode == 501 ? SpanStatus.Unimplemented :
+                    Context.Response.StatusCode == 503 ? SpanStatus.Unavailable :
+                    Context.Response.StatusCode == 504 ? SpanStatus.DeadlineExceeded :
+                    Context.Response.StatusCode < 600 ? SpanStatus.InternalError :
+                    SpanStatus.UnknownError;
+
+                transaction?.Finish(status);
+            }
+        }
+    }
+}
+```
+
+Transaction names follow the pattern `<HTTP method> <Route>`; for example, a request to the following action will create a transaction named `GET /person/{id}`:
+
+```csharp
+using System;
+using System.Web.Mvc;
+
+public class HomeController : Controller
+{
+    [HttpGet("/person/{id}")]
+    public ActionResult Person(string id) { /* ... */ }
+}
+```
+
+Transactions created by this integration are automatically added to the current scope. If you want to start your own spans from the automatically created transaction, you can use the following approach:
+
+```csharp
+using System;
+using System.Web.Mvc;
+using Sentry;
+
+public class HomeController : Controller
+{
+    [HttpGet("/person/{id}")]
+    public ActionResult Person(string id)
+    {
+        var childSpan = SentrySdk.GetSpan()?.StartChild("additional-work");
+        try
+        {
+            // Do the work that gets measured.
+
+            childSpan?.Finish(SpanStatus.Ok);
+        }
+        catch (Exception e)
+        {
+            childSpan?.Finish(SpanStatus.InternalError);
```

Comment:
Don't we have a way to pass the `e` to `Finish` here and find it through the weakref map later to connect the error to the span?

Reply:
Yes